photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/adapter/DevicesAdapter.java

```diff
@@ -35,47 +37,34 @@
  * The implementation of the {@link BaseAdapter} class. Used as an adapter class for the devices list view.
  * Provides list item views with the information about remote devices.
  */
-public class DevicesAdapter extends BaseAdapter {
+public class DevicesAdapter extends ArrayAdapter<DeviceInfo> {
 
     private Context mContext;
-    private KaaManager manager;
-    private List<DeviceInfo> devices;
+    private KaaManager mKaaManager;
+
+    private final LayoutInflater mLayoutInflater;
 
     public DevicesAdapter(Context context, KaaManager controller, List<DeviceInfo> devices) {
```

Comment:
context is not really needed here because you can get context in getView from contentView.getContext()